### PR TITLE
chore: migration of self exclusion and tests

### PR DESCRIPTION
**Pull request by sanjam-deriv**
## Changes:

Migrated self exclusion to typescript
Write tests.
Removed redundant wrapper from sections folder in accounts 

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { act } from 'react-dom/test-utils';
 import { fireEvent, render, screen, waitFor } from '@testing-library/react';
-import SelfExclusion from '../self-exclusion';
+import SelfExclusion from '..';
```
Comment (shaheer-deriv) [inline packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx]: is this change necessary? in other test files, i think we are following this pattern:
`import SelfExclusion from '../self-exclusion';`

File: packages/account/build/webpack.config.js [hunk 2]
```diff
@@ -53,7 +53,6 @@ module.exports = function (env) {
             'poi-poa-docs-submitted': 'Components/poi-poa-docs-submitted/poi-poa-docs-submitted',
             'reset-trading-password-modal': 'Components/reset-trading-password-modal',
             'risk-tolerance-warning-modal': 'Components/trading-assessment/risk-tolerance-warning-modal.jsx',
-            'self-exclusion': 'Components/self-exclusion',
```
Comment (shaheer-deriv) [inline packages/account/build/webpack.config.js:56]: why is this removed?

**Review by shaheer-deriv — APPROVED** [review]

File: packages/account/build/webpack.config.js (hunk 2 above)
Comment (sanjam-deriv) [reply packages/account/build/webpack.config.js:56]: Removed this because the the component is not being exported anywhere outside of the package

File: packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx (hunk 1 above)
Comment (sanjam-deriv) [reply packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx]: It was not reading the file for the some reason when I declared it like that

File: packages/account/src/Sections/Security/SelfExclusion/self-exclusion.tsx [hunk 3]
```diff
@@ -72,13 +72,7 @@ type TResponse = {
...
 };
 
-const SelfExclusion = ({
-    footer_ref,
-    is_app_settings,
-    is_appstore,
-    overlay_ref,
-    setIsOverlayShown,
-}: TSelfExclusion) => {
+const SelfExclusion = ({ footer_ref, is_app_settings, overlay_ref, setIsOverlayShown }: TSelfExclusion) => {
```
Comment (amina-deriv) [inline packages/account/src/Sections/Security/SelfExclusion/self-exclusion.tsx]: SelfExclusion is rendered from route. isnt?

can you double check if these props are valid?

File: packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx (hunk 1 above)
Comment (amina-deriv) [reply packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx]: can you double check this

File: packages/account/build/webpack.config.js (hunk 2 above)
Comment (matin-deriv) [reply packages/account/build/webpack.config.js:56]: @sanjam-deriv does this affect the code splitting dynamics, can you check if the bundle will be created separately or not?

File: packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx (hunk 1 above)
Comment (matin-deriv) [reply packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx]: @sanjam-deriv Can you try wrapping the whole component in observer rather than having this:
export default observer(SelfExclusion);

Comment (sanjam-deriv) [reply packages/account/src/Sections/Security/SelfExclusion/_tests_/self-exclusion.spec.tsx]: Changed this

File: packages/account/build/webpack.config.js (hunk 2 above)
Comment (sanjam-deriv) [reply packages/account/build/webpack.config.js:56]: Added the new route in the config

File: packages/account/src/Sections/Security/SelfExclusion/self-exclusion.tsx [hunk 4]
```diff
@@ -469,12 +457,12 @@ const SelfExclusion = ({
             <SelfExclusionWrapper>
                 {/* Only show the modal in non-"<AppSettings>" views, others will
                     use the overlay provided by <AppSettings> */}
-                {!is_app_settings && <SelfExclusionModal />}
+                {<SelfExclusionModal />}
```
Comment (amina-deriv) [inline packages/account/src/Sections/Security/SelfExclusion/self-exclusion.tsx]: `<SelfExclusionModal />`

File: packages/account/src/Sections/Security/SelfExclusion/self-exclusion.tsx [hunk 5]
```diff
@@ -469,12 +457,12 @@ const SelfExclusion = ({
             <SelfExclusionWrapper>
                 {/* Only show the modal in non-"<AppSettings>" views, others will
                     use the overlay provided by <AppSettings> */}
```
Comment (amina-deriv) [inline packages/account/src/Sections/Security/SelfExclusion/self-exclusion.tsx]: since we removed is_app_settings check, we can remove this comment

**Review by amina-deriv — APPROVED** [review]

File: packages/account/src/Components/self-exclusion/self-exclusion-article.tsx [hunk 6]
```diff
@@ -9,12 +9,11 @@ import SelfExclusionContext from './self-exclusion-context';
 const SelfExclusionArticle = () => {
     const { is_app_settings, toggleArticle, is_eu, is_uk } = React.useContext(SelfExclusionContext);
     const { is_deriv_crypto } = React.useContext(PlatformContext);
-    const { is_appstore } = React.useContext(PlatformContext);
     return (
         <AccountArticle
             title={localize('Trading limits and self-exclusion')}
             descriptions={
-                is_appstore && isDesktop()
+                isDesktop()
```
Comment (suisin-deriv) [inline packages/account/src/Components/self-exclusion/self-exclusion-article.tsx]: can you pull the latest changes, we have included `is_desktop` in ui we can use that instead of isDesktop 😎

File: packages/account/src/Components/self-exclusion/self-exclusion-article.tsx [hunk 7]
```diff
@@ -37,7 +36,7 @@ const SelfExclusionArticle = () => {
                           ),
                       ]
             }
-            {...((!is_appstore || isMobile()) && { onClickLearnMore: toggleArticle })}
+            {...(isMobile() && { onClickLearnMore: toggleArticle })}
```
Comment (suisin-deriv) [inline packages/account/src/Components/self-exclusion/self-exclusion-article.tsx]: same goes to `is_mobile`

File: packages/account/src/Components/self-exclusion/self-exclusion-article.tsx [hunk 8]
```diff
@@ -1,20 +1,22 @@
...
 import { StaticUrl } from '@deriv/components';
 import { Localize, localize } from '@deriv/translations';
-import { getBrandWebsiteName, getPlatformSettings, PlatformContext, isDesktop, isMobile } from '@deriv/shared';
+import { getBrandWebsiteName, getPlatformSettings, PlatformContext } from '@deriv/shared';
 import AccountArticle from 'Components/article';
 import { selfExclusionArticleItems } from 'Components/self-exclusion/self-exclusion-article-content';
 import SelfExclusionContext from './self-exclusion-context';
+import { useStore } from '@deriv/stores';
 
 const SelfExclusionArticle = () => {
```
Comment (suisin-deriv) [inline packages/account/src/Components/self-exclusion/self-exclusion-article.tsx]:
```suggestion
const SelfExclusionArticle = observer(() => {
```

**Review by suisin-deriv — APPROVED** [review]
